pkg/pullrequest/scm.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pullrequest
+
+import (
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/jenkins-x/go-scm/scm/driver/github"
+	"github.com/jenkins-x/go-scm/scm/driver/gitlab"
+	"go.uber.org/zap"
+)
+
+func NewSCMHandler(logger *zap.SugaredLogger, raw string) (*Handler, error) {
+	u, err := url.Parse(raw)
+	if err != nil {
+		return nil, err
+	}
+
+	switch {
+	case strings.Contains(raw, "github"):
```

Comment:
This isn't going to be a reliable way to determine the type. e.g. This would match something like `https://gitlab.com/tektoncd/github-compat/merge_requests/1`

For simplicity, switch on `u.Host`. We'll likely need to add a flag for SCM type, since we're not going to be able to do this for enterprise hosts.

Reply:
Ah yeah, great point. I'll add that in here. The idea will be: use the flag if it's set, otherwise try to guess.